webxr-api/session.rs

```diff
@@ -42,6 +42,42 @@ pub enum SessionMode {
     ImmersiveAR,
 }
 
+/// https://immersive-web.github.io/webxr/#dictdef-xrsessioninit
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "ipc", derive(Serialize, Deserialize))]
+pub struct SessionInit {
+    pub required_features: Vec<String>,
+    pub optional_features: Vec<String>,
+}
+
+impl SessionInit {
+    /// Helper function for validating a list of requested features against
+    /// a list of supported features for a given mode
+    pub fn validate(&self, mode: SessionMode, supported: &[String]) -> Result<Vec<String>, Error> {
+        for f in &self.required_features {
+            // viewer and local in immersive are granted by default
+            if f == "viewer" || (f == "local" && mode != SessionMode::Inline) {
+                continue;
+            }
+
+            if !supported.contains(f) {
+                return Err(Error::UnsupportedFeature(f.into()));
```

Comment:
This will only report an error for the first missing feature, is that what's wanted?

Reply:
Yeah, JS doesn't see this error anyway, and the spec doesn't expose exactly what features were unsupported (but I have a PR to make it expose what features were _granted_ )